flexboxAPIspecV2.md

```diff
@@ -0,0 +1,783 @@
+
+
+
+# API Pages
+
+_(Each of the following L2 sections correspond to a page that will be on docs.microsoft.com)_
+
+<!-- TEMPLATE
+
+  Each of the L2 sections in this "API Pages" section corresponds to a page on DMC.
+
+  It's not necessary to have a section for every class member though:
+  * If its purpose and usage is obvious from it's name/type, it's not necessary to
+    create a section for it.
+  * If its purpose and usage is fully explained by brief description, either
+      put it in a table in the "Other [class] members" section
+      put it with /// comments in the IDL section
+
+  Create an L2 section here for each API that needs more description or examples.
+  For a new class with members, the members should go in their own L2 section.
+
+  Example layout
+    ## MyClass
+    ## MyClass.Member1
+    ## MyClass.Member2
+    ## Other MyClass members
+    ## MyOtherClass
+    ## ...
+
+  Notes:
+  * The first line of each of these sections should become that first line on the DMC page,
+    which then becomes the description you see in Intellisense.
+  * Each page can have description, examples, and remarks.
+    Remarks are where the documentation calls out special considerations that the developer
+    should be aware of.
+  * It can be helpful at the top of an API page (or after the Intellisense text) to add the
+    API signature in C#
+  * Add a "_Spec note: ..._" to add a note that's useful in this spec but shouldn't go to DMC.
+  * Show _examples_, not _samples_; an example is a snippet, a sample is a full working app.
+
+-->
+
+## Flex Direction
+
+
+The flex-direction property specifies the direction of the flexible items.
+Note: If the element is not a flexible item, the flex-direction property has no effect.
+
+
+```xml
+<controls:LayoutPanel>
+    <controls:LayoutPanel.Layout>
+        <controls:FlexboxLayout Direction="Row"/>
+    </controls:LayoutPanel.Layout>
+            <Grid>
+                    <TextBlock>1</TextBlock>
+            </Grid>
+            <Grid>
+                <TextBlock>2</TextBlock>
+            </Grid>
+            <Grid>
+                  <TextBlock>3</TextBlock>
+            </Grid>
+            <Grid>
+                  <TextBlock>4</TextBlock>
+            </Grid>
+</controls:LayoutPanel>
+```
+
+
+
+### Output of the above code (styling omitted)
+![image](./FlexDirectionRow.PNG)
+
+
+
+##  Flex Direction Accepted Values
+
+| Value | Description |
+|-|-|
+| Row (Default) | The flexible items are displayed horizontally, as a row |
+| RowReverse | Same behavior as Row, but elemnts are shown in reverse |
+| Column | The flexible items are displayed vertically, as a column |
+| ColumnReverse | Same behavior as Column, but elemnts are shown in reverse |
+
+# API Details
+
+```c# (but really MIDL3)
+
+runtimeclass FlexboxLayout;
+
+    enum FlexboxDirection
```

Comment:
In the code snippet "Direction" is the property name, in the enum it is "FlexboxDirection". Which name is going to be used?

Reply:
It's OK for the property name to be shorter than the type name, since member names are always in the context of a class. So I think it looks good.